src/components/fields/location-field/location-field.tsx

```diff
@@ -24,13 +28,52 @@ export const LocationField = (props: IGenericFieldProps<ILocationFieldSchema>) =
 			gettingCurrentLocationFetchMessage,
 			mustHavePostalCode,
 			locationModalStyles,
+			validation,
 		},
 		// form values can initially be undefined when passed in via props
 		value: formValue,
 		onChange,
+		error,
 	} = props;
 
 	const [showLocationModal, setShowLocationModal] = useState<boolean>(false);
+	const { setFieldValidationConfig } = useValidationConfig();
+
+	// =============================================================================
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
+		const isRequiredRule = validation?.find((rule) => "required" in rule);
+
+		setFieldValidationConfig(
+			id,
+			Yup.object({
+				address: Yup.string(),
+				blockNo: Yup.string(),
+				building: Yup.string(),
+				lat: Yup.number(),
+				lng: Yup.number(),
+				postalCode: Yup.string(),
+				roadName: Yup.string(),
+				x: Yup.number(),
+				y: Yup.number(),
+			})
+				.test("is-required", isRequiredRule?.errorMessage || ERROR_MESSAGES.COMMON.FIELD_REQUIRED, (value) => {
+					if (!isRequiredRule?.required) return true;
+					return !!value && Object.values(value).filter((val) => !!val).length > 0;
+				})
+				.test("must-have-postal-code", ERROR_MESSAGES.LOCATION.MUST_HAVE_POSTAL_CODE, (value) => {
+					if (!mustHavePostalCode) return true;
+					return LocationHelper.hasGotAddressValue(value.postalCode);
+				})
+				.test("must-have-location-data", ERROR_MESSAGES.LOCATION.INVALID_LOCATION, (value) => {
+					if (!value) return true;
+					return !!value.x && !!value.y && !!value.lat && !!value.lng;
+				}),
```

Comment:
can we combine this check with `is-required`? because this is preventing submission even though no validation config is specified (aka optional field)